horovod/tensorflow/__init__.py

```diff
@@ -495,6 +496,8 @@ def compute_gradients(self, *args, **kwargs):
             """
             gradients = self._optimizer.compute_gradients(*args, **kwargs)
             grads, vars = zip(*gradients)
+            if self._group is not None and not isinstance(self._groups, list) and self._group > 0:
```

Comment:
Could you motivate the logic in this condition a bit?

`self._group` is probably a typo for `self._groups`?

Is this fix not necessary for the case where `self._groups` is a list?

Reply:
Sorry for the typo here, I have corrected it. 
Yes, I think this fix is unnecessary for the case where self._groups is a list. In this case (self._groups: list) the code handles both var name and grad, which will not lead to the error discribed in the issue. 
Next I will try to add a unit test to test_tensorflow.py to reproduce the problem.